src/core/interpreter/profile-parameter-validator.test.ts

```diff
@@ -22,1005 +20,768 @@ const checkErrorKind = (result: Result<unknown, unknown>) =>
       });
 
-      it('should pass with valid input', () => {
-        expect(
-          parameterValidator
-            .validate({ test: { hello: 'world!' } }, 'input', 'Test')
-            .isOk()
-        ).toEqual(true);
+      it('returns ok for foo not being passed', () => {
+        expect(parameterValidator.validate({}, 'result', 'Test').isOk()).toBeTruthy();
       });
 
-      it('should fail with invalid input', () => {
-        const result1 = parameterValidator.validate(
-          { test: 'hello!' },
-          'input',
-          'Test'
-        );
-        expect(checkErrorKind(result1)).toEqual(['wrongType']);
-        expect(checkErrorPath(result1)).toEqual([['input', 'test']]);
-        const result2 = parameterValidator.validate(
-          { test: {} },
-          'input',
-          'Test'
-        );
-        expect(checkErrorKind(result2)).toEqual(['missingRequired']);
-        expect(checkErrorPath(result2)).toEqual([['input', 'test', 'hello']]);
-        const result3 = parameterValidator.validate(
-          { test: { hello: 7 } },
-          'input',
-          'Test'
-        );
-        expect(checkErrorKind(result3)).toEqual(['wrongType']);
-        expect(checkErrorPath(result3)).toEqual([['input', 'test', 'hello']]);
+      it('returns ok for foo being undefined', () => {
+        expect(parameterValidator.validate({ foo: undefined }, 'result', 'Test').isOk()).toBeTruthy();
       });
     });
 
-    describe('AST with union', () => {
+    describe('required fields', () => {
       const ast = parseProfileFromSource(`
-        usecase Test safe {
-          input {
-            test string | number
+        usecase Test {
+          result {
+            foo! string
           }
         }
       `);
-      let parameterValidator: ProfileParameterValidator;
 
       beforeEach(() => {
         parameterValidator = new ProfileParameterValidator(ast);
       });
 
-      it('should pass with valid input', () => {
-        expect(parameterValidator.validate({}, 'input', 'Test').isOk()).toEqual(
-          true
-        );
-        expect(
-          parameterValidator.validate({ test: 'hello' }, 'input', 'Test').isOk()
-        ).toEqual(true);
-        expect(
-          parameterValidator.validate({ test: 7 }, 'input', 'Test').isOk()
-        ).toEqual(true);
+      it('returns error for foo not being passed', () => {
+        expect(parameterValidator.validate({}, 'result', 'Test').isErr()).toBeTruthy();
       });
 
-      it('should fail with invalid input', () => {
-        const result = parameterValidator.validate(
-          { test: true },
-          'input',
-          'Test'
-        );
-        expect(checkErrorKind(result)).toEqual(['wrongUnion']);
-        expect(checkErrorPath(result)).toEqual([['input', 'test']]);
+      it('returns error for foo being undefined', () => {
+        expect(parameterValidator.validate({ foo: undefined }, 'result', 'Test').isErr()).toBeTruthy();
```

Comment:
Well we need to clear one thing. Are we adding `None` as type now, or continue with undefined and null as they were used now?

Reply:
Because we agreed to keep it internal and not surface right now.

Reply:
Well the branch is called `feat/none_type`, so I was assuming we are in the context of adding it as a type. It doesn't have to be a type visible in the profile (I used that for illustrative purposes, we could make foo an untyped field as well), but we are defining the semantics of a None type, and these semantics will be observable in sdk.

Reply:
As I understand it we aren't surfacing None type as an explicit name, but we are defining how it will behave and that will have user-visible consequences through sdk behavior.